src/solver/optimisation/adequacy_patch_csr/solve_problem.cpp

```diff
@@ -161,7 +150,7 @@ void storeOrDisregardInteriorPointResults(const PROBLEME_ANTARES_A_RESOUDRE* Pro
           << ". hour: " << weekNb * hoursInWeek + hourlyCsrProblem.hourInWeekTriggeredCsr + 1;
 }
 
-double calculateCsrCostFunctionValue(const PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre,
+double calculateCsrCostFunctionValue(const std::unique_ptr<PROBLEME_POINT_INTERIEUR>& Probleme,
```

Comment:
Design suggestion, from the c++ core guidelines:
we should use `const PROBLEME_POINT_INTERIEUR&` as a parameter here (or `PROBLEME_POINT_INTERIEUR&`, if modified). Using a std::unique_ptr should be reserved to transferring ownership, and there is no point in requiring a unique_ptr here instead of another kind of pointer/reference.

Reference is the right choice compared to plain pointer because here we expect a non-null value.

Reply:
[done](088a498f)